crates/sdk/src/build.rs

```diff
@@ -96,6 +100,22 @@ impl Drop for TarFile {
     }
 }
 
+struct CountingReader<R: Read> {
+    inner: R,
+    progress: Arc<AtomicU64>,
+}
+
+impl<R: Read> Read for CountingReader<R> {
```

Comment:
Would be nice to move it to common (maybe just `lib.rs` or a new separate file) and reuse it everywhere. I am not confident we use atomics btw (see comment about thread below)